hibernate-core/src/main/java/org/hibernate/query/sqm/produce/function/internal/PatternRenderer.java

```diff
@@ -94,14 +101,16 @@ else if ( c  == '?' ) {
 			chunkList.add( chunk.toString() );
 		}
 
-		varargParam = vararg;
-		maxParamIndex = max;
+		this.varargParam = vararg;
+		this.maxParamIndex = max;
```

Comment:
why the useless `this`s?

Reply:
For consistency sake as I added a local variable for `paramIndexes` to only set the final variable `this.paramIndexes` once the array is fully initialized.